docs/source/tutorial/cli/hpopt.rst

```diff
@@ -0,0 +1,74 @@
+
+ * :code:`depth` The number of message passing steps
+ * :code:`ffn_num_layers` The number of layers in the FFN model
+ * :code:`dropout` The probability (from 0 to 1) of dropout in the MPNN & FNN layers
+ * :code:`message_hidden_dim` The hidden dimension in the message passing step 
+ * :code:`ffn_hidden_dim` The hidden dimension in the FFN model
+
+Another option is :code:`learning_rate` which includes:
+
+ * :code:`max_lr` The maximum learning rate
+ * :code:`init_lr_ratio` Ratio of the initial learning rate to the max learning rate
+ * :code:`final_lr_ratio` Ratio of the final learning rate to the max learning rate 
+ * :code:`warmup_epochs` Number of warmup epochs, during which the learning rate linearly increases from the initial to the maximum learning rate
+
+Other individual search parameters include:
+
+ * :code:`activation` The activation function used in the MPNN & FFN layers. Choices include ``relu``, ``leakyrelu``, ``prelu``, ``tanh``, ``selu``, and ``elu``
+ * :code:`aggregation` Aggregation mode used during molecule-level predictor. Choices include ``mean``, ``sum``, ``norm``
+ * :code:`aggregation_norm` For ``norm`` aggregation, the normalization factor by which atomic features are divided
+ * :code:`batch_size` Batch size for dataloader
+
+Specifying :code:`--search-parameter-keywords all` will search over all 13 of the above parameters.
+
+The following other common keywords may be used:
+ 
+ * :code:`--raytune-num-samples <num_samples>` The number of conditions to sample 
+ * :code:`--raytune-num-cpus <num_cpus>` The number of CPUs to use  
+ * :code:`--raytune-num-gpus <num_gpus>` The number of GPUs to use  
+ * :code:`--raytune-max-concurrent-trials <num_trials>` The maximum number of concurrent trials
+ * :code:`--raytune-search-algorithm <algorithm>` The choice of control search algorithm (either ``random``, ``hyperopt``, or ``optuna``). If ``hyperopt`` is specified, then the arguments ``--hyperopt-n-initial-points <num_points>`` and ``--hyperopt-random-state-seed <seed>`` can be specified.
+
+Other keywords related to hyperparameter optimization are also available (see :ref:`cmd` for a full list).
+
+Applying Optimal Hyperparameters
+---------------------------------
+
+Once hyperparameter optimization is complete, the optimal hyperparameters can be applied during training by specifying the config path. For example:
+
+.. code-block::
+
+   chemprop train --data-path tests/data/regression.csv \
+   --task-type regression \
+   --config-path results/best_config.toml \
+
+Note that the hyperparameter optimization script sees all the data given to it. The intended use is to run the hyperparameter optimization script on a dataset with the eventual test set held out. 
+
+.. If you need to optimize hyperparameters separately for several different cross validation splits, you should e.g. set up a bash script to run :code:`chemprop hpopt` separately on each split's training and validation data with test held out.
```

Comment:
```suggestion
.. note::
    If you need to optimize hyperparameters separately for several different cross validation splits, you should e.g. set up a bash script to run :code:`chemprop hpopt` separately on each split's training and validation data with test held out.
```

Reply:
I had left this commented out from v1 as I wasn't sure if this was still the recommended method, is this still correct?

Reply:
After reviewing the code, we use only the first split to perform HPO when `--num-folds` is greater than 1 (see [here](https://github.com/chemprop/chemprop/blob/main/chemprop/cli/hpopt.py#L485-L486)). Therefore, if one wants to optimize hyperparameters across different splits, they should provide their own data splits.
@hwpang we should add a sentence in the documentation and also a message in the CLI to let people know that only the first split is used.

Reply:
okay, will add this as well